src/python_testing/conformance_support.py

```diff
@@ -58,14 +87,27 @@ class ConformanceDecision(Enum):
     PROVISIONAL = auto()
 
 
+@dataclass
+class ConformanceDecisionWithChoice:
+    decision: ConformanceDecision
+    choice: Optional[Choice] = None
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
```

Comment:
A comment saying "this supports comparison to ConformanceDecision without considering the choice" seems maybe useful as this seems to be the important point.

Maybe also `@dataclass(eq=false)` to make it clear we do not want `__eq__` autogenerated by dataclass.

Reply:
This also is probably tricky because `a == b` does not imply `b == a` ... which seems kind of dangerous.

How about removing `__eq__` and implementing a `SameDecision(self, other)` to make the ordering requirement explicit?

Reply:
Removed this and just compared directly - you're right, it's more maintainable etc.